pkg/containerruntime/docker_config.go

```diff
@@ -32,12 +33,17 @@ type dockerConfig struct {
 }
 
 func marshalDockerConfig(cluster *kubeone.KubeOneCluster) (string, error) {
+	logSize := strings.ToLower(cluster.LoggingConfig.ContainerLogMaxSize)
+	logSize = strings.ReplaceAll(logSize, "ki", "k")
+	logSize = strings.ReplaceAll(logSize, "mi", "m")
```

Comment:
ReplaceAll is case sensitive, it will not replace `Mi` with `m`, but instead would simply return `Mi` after which dockerd would freak out.